mod/groups/start.php

```diff
@@ -680,41 +695,6 @@ function groups_get_invited_groups($user_guid, $return_guids = false, $options =
 }
 
 /**
- * Join a user to a group, add river event, clean-up invitations
- *
- * @param ElggGroup $group
- * @param ElggUser  $user
- * @return bool
- */
-function groups_join_group($group, $user) {
-
-	// access ignore so user can be added to access collection of invisible group
-	$ia = elgg_set_ignore_access(TRUE);
```

Comment:
i tested the scenario described in the ticket... the problem is back... so there still needs to be done something special with the entity (maybe not as part of join) in the invite action